File: backend/turip-app/src/main/java/turip/article/controller/dto/response/ArticlesResponse.java
package turip.article.controller.dto.response;

import java.util.List;

public record ArticlesResponse(
        List<ArticleSummaryResponse> articles,
        boolean loadable
) {

    public static ArticlesResponse of(List<ArticleSummaryResponse> articles, boolean loadable) {
        return new ArticlesResponse(articles, loadable);
    }
}

File: mobile/core/data/src/commonMain/kotlin/com/on/turip/core/data/dto/article/ArticlesResponse.kt
package com.on.turip.core.data.dto.article

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ArticlesResponse(
    @SerialName("articles")
    val articles: List<ArticleResponse>,
    @SerialName("loadable")
    val loadable: Boolean,
)
